### PR TITLE
checking return values for incoming check

### LINKED ISSUE #2843: python tools - --incoming check does not return the correct exit value
```
ktulinger:opengrok ktulinger$ opengrok-mirror --loglevel INFO -c /var/opengrok/etc/mirror-config-generated.yml --incoming -U http://localhost:8080 project-x
output of ['/usr/local/bin/git', 'pull', '--dry-run']:

No incoming changes for repositories in project project-x
ktulinger:opengrok ktulinger$ echo $?
0
```

As I understood the return value should be 2, so the sync chain is skipped however this is never met.

### PR DESCRIPTION
**Pull request by tulinkry**
- refactoring the code to use the constants

fixes #2843

### CODE REVIEW

File: opengrok-tools/src/main/python/opengrok_tools/mirror.py [hunk 1]
```diff
@@ -177,11 +182,13 @@ def main():
                 except KeyboardInterrupt:
                     sys.exit(1)
                 else:
-                    if any([True for x in project_results if x == 1]):
-                        ret = 1
+                    if any([True for x in project_results if x == FAILURE_EXITVAL]):
+                        ret = FAILURE_EXITVAL
+                    if all([True for x in project_results if x == CONTINUE_EXITVAL]):
```
Comment (vladak) [inline opengrok-tools/src/main/python/opengrok_tools/mirror.py]: I think you need to check indeed for all. Something like:
```python
(len([x for x in project_results if x == 2]) == len(project_results))
```

Comment (vladak) [reply opengrok-tools/src/main/python/opengrok_tools/mirror.py]: or something like:
```python
 all(list(map(lambda x: True if x == 1 else False, project_results)))
```

**Comment by vladak** [comment]
There is the logic problem + style issues, otherwise this is good to go.

File: opengrok-tools/src/main/python/opengrok_tools/mirror.py (hunk 1 above)
Comment (tulinkry) [reply opengrok-tools/src/main/python/opengrok_tools/mirror.py]: yes I discovered the problem also

Comment (tulinkry) [reply opengrok-tools/src/main/python/opengrok_tools/mirror.py]: I went for `all([x == CONTINUE_EXITVAL for x in project_results])`

Comment (tulinkry) [reply opengrok-tools/src/main/python/opengrok_tools/mirror.py]: which is essentially your last suggestion

**Comment by tulinkry** [comment]
How about now?

**Pull request merged**